otlp/trace.go

```diff
@@ -80,19 +80,13 @@ func TranslateTraceRequest(request *collectorTrace.ExportTraceServiceRequest, ri
 		if isLegacy {
 			dataset = ri.Dataset
 		} else {
-			if resourceSpan.Resource == nil {
```

Comment:
simplified the branching here: 
- `resourceAttrs` will just be empty if `resourceSpan.Resource` is nil, so no need to pre-check
- collapsed ifs and elses so there one default case and one main case